modules/saas/apps/saml-saas/src/main/java/com/liferay/saml/saas/internal/portlet/filter/SamlAdminRenderFilter.java

```diff
@@ -39,16 +42,22 @@
 import javax.servlet.RequestDispatcher;
 import javax.servlet.ServletContext;
 
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.component.annotations.Activate;
 import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
 import org.osgi.service.component.annotations.Reference;
 
 /**
  * @author Marta Medio
  */
 @Component(
+	configurationPid = "com.liferay.saml.runtime.configuration.SamlKeyStoreManagerConfiguration",
 	immediate = true,
 	property = "javax.portlet.name=" + SamlAdminPortletKeys.SAML_ADMIN,
-	service = PortletFilter.class
+	service = RenderFilter.class
 )
 public class SamlAdminRenderFilter implements RenderFilter {
```

Comment:
If we are only using this component to conditionally register another service, I don't think we need to register it as a RenderFilter service? Preferability it would be a component without any service.
Also shouldn't we call it SamlAdminRenderFilterRegistrator?

Reply:
`formatSource` dynamically adds `service = RenderFilter.class`, it does not allow us to have this component without indicating that line 😩 

I'm fine with the name change

Reply:
I see, didn't know that. Maybe we should then register it as a service of the concrete class (nothing is tracking that)?

Reply:
have you tried with `service = {}`?

Reply:
That's it 👏 
I just pushed the change.

Reply:
:clap: